virttest/utils_test/__init__.py

```diff
@@ -1947,149 +1951,148 @@ def _tarball_download(self, url, destination):
-        self.host_stress_process = subprocess.Popen(args, universal_newlines=True,
-                                                    stdout=subprocess.PIPE,
-                                                    stderr=subprocess.PIPE)
-        running = utils_misc.wait_for(self.app_running, first=0.5, timeout=60)
-        if not running:
-            raise StressError("Stress tool %s isn't running"
-                              % self.stress_type)
-
-    @error_context.context_aware
-    def unload_stress(self):
-        """
-        stop stress tool manually
-        """
-        def _unload_stress():
-            if self.host_stress_process is not None:
-                utils_misc.kill_process_tree(self.host_stress_process.pid)
-            if not self.app_running():
-                return True
-            return False
+            session = self.vm.wait_for_login()
+            return session
+        except aexpect.ShellError as detail:
+            raise StressError("Login %s failed:\n%s" % (self.vm.name, detail))
 
-        error_context.context("stop stress app on host", logging.info)
-        utils_misc.wait_for(_unload_stress, first=2.0,
-                            text="wait stress app quit", step=1.0, timeout=60)
 
-    def app_running(self):
+class HostStress(Stress):
+
+    """
+    Run Stress tool on host, such as stress, unixbench, iozone and etc.
+    """
+
+    def __init__(self, stress_type, params, download_url="", make_cmds="",
+                 stress_cmds="", stress_args="", work_path="",
+                 uninstall_cmds="", download_type="tarball"):
+        super(HostStress, self).__init__(stress_type, params, download_url="",
+                                         make_cmds="", stress_cmds="", stress_args="",
+                                         work_path="", uninstall_cmds="",
+                                         download_type="tarball")
+        remote_ip = params.get("remote_ip", None)
+        remote_pwd = params.get("remote_pwd", None)
+        remote_user = params.get("remote_user", "root")
+        if remote_ip and remote_pwd:
+            self.remote_host = {'server_ip': remote_ip, 'server_pwd': remote_pwd}
+            self.remote_host['server_user'] = remote_user
+            self.copy_files_to = remote.copy_files_to
+            self.session = self.get_session()
+            self.cmd_status_output = self.session.cmd_status_output
+            self.cmd_status = self.session.cmd_status
+            self.cmd_output_safe = self.session.cmd_output_safe
+            self.sendline = self.session.sendline
+
+    def get_session(self):
         """
-        check whether app really run in background
+        Method to get the session of the remote host
         """
-        result = process.run(self.check_cmd, timeout=60, ignore_status=True)
-        return result.exit_status == 0
+        from virttest import test_setup
+        try:
+            session = test_setup.remote_session(self.remote_host)
```

Comment:
Looks like this has to be a remote host, `self.remote_host` wont be defined if no `remote_ip` and no `remote_pwd` defined. Will this works for local host?

Reply:
It will work for local host, by default the commands are executed using process cmds
```python
        self.cmd_status_output = process.getstatusoutput
        self.cmd_status = process.system
        self.cmd_output_safe = self.sendline = process.getoutput
```

Reply:
@balamuruhans Oh I see, sorry I missed it.